### DISCUSSION TITLE
How to conditionally register / unregister a required field based on another form value ?

### DISCUSSION THREAD

**Discussion by gowthamvbhat** [react-hook-form/react-hook-form#7542] [Q&A] [answered]
Hi, I have a form where one of the fields become active only when a checkbox is enabled. 
But when enabled, it is a **required** field.

A simple example is a form with 3 fields.
- Product Name
- Primary cost
- Secondary Cost

By default, secondary cost is calculated in the back-end after form submission.
But an option is provided to override the secondary cost and enter it manually.

The secondary cost field is enabled / disabled using a checkbox. (The value of this field is not relevant in the back-end)

Ideally when the secondary cost field is disabled, it should not be registered in the form or at least validation should not be carried out for that field.

[Example Sandbox](https://codesandbox.io/s/react-hook-form-register-forked-nd80b?file=/src/index.js)

Is there a way to achieve this ?

Thanks in advance!

**Comment by Moshyfawn** [comment, marked as answer]
What you want is [the `unregister` method](https://www.react-hook-form.com/api/useform/unregister/), I believe. See [this codesandbox](https://codesandbox.io/s/rhf-unregister-example-1qi7s).
Take note of this rule:
> Make sure you unmount that input which has register callback or else the input will get registered again.

In your case, leaving the input mounted when disabled will result in re-registering it back with all it's validation, if it gets enabled again. So, no need to do anything else other than unregister it when disabled.

**Comment by gowthamvbhat** [comment]
@Moshyfawn Thanks a lot

**Reply by gowthamvbhat** [reply]
Found something interesting when trying this out and wanted to share here.

I actually had two fields for secondary cost,
- value registered as 'secondaryCost.value'
- currency registered as 'secondaryCost.currency' 

While un-registering I just had to do unregister('secondaryCost') and both fields were taken care of!

This is awesome!

**Reply by Moshyfawn** [reply]
Yep, 'cuz RHF is awesome like that! xd
Thx for sharing!
